src/System.Management.Automation/engine/CommandSearcher.cs

```diff
@@ -456,52 +456,28 @@ private CommandInfo GetNextFromPath()
                     "Trying to resolve the path as an PSPath");
 
                 // Find the match if it is.
-
-                Collection<string> resolvedPaths = new Collection<string>();
-
-                try
-                {
-                    Provider.CmdletProvider providerInstance;
-                    ProviderInfo provider;
-                    resolvedPaths =
-                        _context.LocationGlobber.GetGlobbedProviderPathsFromMonadPath(_commandName, false, out provider, out providerInstance);
-                }
-                catch (ItemNotFoundException)
+                // Try literal path resolution if it is set to run first
+                if (_commandResolutionOptions.HasFlag(SearchResolutionOptions.ResolveLiteralThenPathPatterns))
```

Comment:
if `ResolveLiteralThenPathPatterns` is set, then the code flow will return from within this if block.